File: bluetooth/src/main/java/com/ankerboxmanager/bluetooth/scaner/LegacyLeBluetoothScanner.java
//package com.ankerboxmanager.bluetooth.scaner;
//
//import com.ankerboxmanager.bluetooth.IBluetooth;
//
//import android.annotation.TargetApi;
//import android.bluetooth.BluetoothAdapter;
//import android.bluetooth.BluetoothDevice;
//import android.bluetooth.BluetoothManager;
//import android.content.Context;
//import android.os.Build;
//
///**
// * Created on 15/05/2018.
// *
// * @author lx
// */
//
//@TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR2)
//public class LegacyLeBluetoothScanner extends BluetoothScanner {
//
//    private BluetoothAdapter.LeScanCallback leScanCallback = new BluetoothAdapter.LeScanCallback() {
//        @Override
//        public void onLeScan(BluetoothDevice device, int rssi, byte[] scanRecord) {
//            onNewDeviceFound(device);
//        }
//    };
//
//    public LegacyLeBluetoothScanner(Context context) {
//        super(context);
//    }
//
//    @Override
//    public void startScan(String[] serviceUUIDs, IBluetooth.AsyncReturnCallback asyncReturnCallback) {
//        boolean result = getBluetoothAdapter().startLeScan(leScanCallback);
//        if (asyncReturnCallback != null) {
//            asyncReturnCallback.onReturn(result);
//        }
//    }
//
//    @Override
//    public void stopScan(IBluetooth.AsyncReturnCallback asyncReturnCallback) {
//        getBluetoothAdapter().stopLeScan(leScanCallback);
//        if (asyncReturnCallback != null) {
//            asyncReturnCallback.onReturn(true);
//        }
//    }
//
//    private BluetoothAdapter getBluetoothAdapter() {
//        BluetoothManager manager = (BluetoothManager) context.getSystemService(Context.BLUETOOTH_SERVICE);
//        return manager.getAdapter();
//    }
//
//}

File: bluetooth/src/main/kotlin/com/ankerboxmanger/bluetooth/scaner/LegacyLeBluetoothScanner.kt
package com.ankerboxmanger.bluetooth.scaner

import android.annotation.TargetApi
import android.bluetooth.BluetoothAdapter
import android.bluetooth.BluetoothManager
import android.content.Context
import android.os.Build
import com.ankerboxmanger.bluetooth.IBluetooth

/**
 * Created on 15/05/2018.
 *
 * @author lx
 */

@TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR2)
class LegacyLeBluetoothScanner(context: Context) : BluetoothScanner(context) {

    private val leScanCallback = BluetoothAdapter.LeScanCallback {
        device, rssi, scanRecord -> onNewDeviceFound(device)
    }

    private val bluetoothAdapter: BluetoothAdapter
        get() {
            val manager = context.getSystemService(Context.BLUETOOTH_SERVICE) as BluetoothManager
            return manager.adapter
        }

    override fun startScan(serviceUUIDs: Array<String>?, asyncReturnCallback: IBluetooth.AsyncReturnCallback?) {
        val result = bluetoothAdapter.startLeScan(leScanCallback)
        asyncReturnCallback?.onReturn(result)
    }

    override fun stopScan(asyncReturnCallback: IBluetooth.AsyncReturnCallback?) {
        bluetoothAdapter.stopLeScan(leScanCallback)
        asyncReturnCallback?.onReturn(true)
    }

}
